plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4265,6 +4267,101 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
+            // @description
+            // There are several tags to locate structures, such as <@link tag LocationTag.find_structure> and <@link tag LocationTag.find_structure_type>.
+            // These tags work similarly to the '/locate' command, and have several side effects/edge cases:
+            // - The radius is in chunks, but isn't always a set square radius around the origin; certain structures may modify the amounts of chunks checked.
+            //   For example, woodland mansions can potentially check up to 20,000 blocks away (or more) regardless of the radius used.
+            // - Lookups can take a long amount of time (several seconds, over 10 in some cases), especially when looking for unexplored structures,
+            //   which will cause the server to freeze while searching.
+            // - They will not load/generate chunks (but can search not-yet-generated chunks and return a location in them).
+            // - They can lead to situations where the server hangs and crashes when trying to find unexplored structures (if there aren't any/any nearby),
+            //   as it keeps looking further and further out.
+            // - The returned location only contains the X and Z values, and will always have a Y value of 0.
+            //   Tags like <@link tag LocationTag.highest> are available, but note that they require the chunk to be loaded.
+            // -->
+
+            // <--[tag]
+            // @attribute <LocationTag.find_structure[structure=<structure>;radius=<number>(;unexplored=<true/{false}>)]>
+            // @returns LocationTag
+            // @warning See <@link language Structure lookups> for potential issues/edge cases in structure lookups.
+            // @group finding
+            // @description
+            // Finds the closet structure of the given type within the specified chunk radius (if any), optionally only searching for unexplored ones.
+            // For a list of default structures, see <@link url https://minecraft.wiki/w/Structure#ID>.
+            // Alternatively, you can specify a custom structure from a datapack, plugin, etc. as a namespaced key.
+            // See also <@link tag LocationTag.find_structure_type> to find structures by type,
+            // and <@link tag server.structures> for all structures currently available on the server.
+            // -->
+            tagProcessor.registerTag(LocationTag.class, MapTag.class, "find_structure", (attribute, object, input) -> {
+                ElementTag structureName = input.getRequiredObjectAs("structure", ElementTag.class, attribute);
+                ElementTag radius = input.getRequiredObjectAs("radius", ElementTag.class, attribute);
+                if (structureName == null || radius == null) {
+                    return null;
+                }
+                org.bukkit.generator.structure.Structure structure = Registry.STRUCTURE.get(Utilities.parseNamespacedKey(structureName.asString()));
```

Comment:
why the long type

Reply:
`org.bukkit.block.Structure` (Structure blocks) is also imported